### PR TITLE
fix multi-attachment deletion issue

### PR DESCRIPTION
**Pull request by gailwang**
create:

`data.tencentcloud_instances.default: Refreshing state...
tencentcloud_clb_attachment.default[0]: Refreshing state... [id=#lbl-g84n3gr2#lb-1sp4y4so]
tencentcloud_clb_attachment.default[1]: Refreshing state... [id=#lbl-g84n3gr2#lb-1sp4y4so]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # tencentcloud_clb_attachment.default[0] will be created
  + resource "tencentcloud_clb_attachment" "default" {
      + clb_id        = "lb-1sp4y4so"
      + id            = (known after apply)
      + listener_id   = "lbl-g84n3gr2"
      + protocol_type = (known after apply)

      + targets {
          + instance_id = "ins-fa2damgc"
          + port        = 80
          + weight      = 10
        }
    }

  # tencentcloud_clb_attachment.default[1] will be created
  + resource "tencentcloud_clb_attachment" "default" {
      + clb_id        = "lb-1sp4y4so"
      + id            = (known after apply)
      + listener_id   = "lbl-g84n3gr2"
      + protocol_type = (known after apply)

      + targets {
          + instance_id = "ins-bvpf6j4w"
          + port        = 80
          + weight      = 10
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Warning: Quoted type constraints are deprecated

  on variable.tf line 3, in variable "instance_ip":
   3:   type = "list"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "list" and write
list(string) instead to explicitly indicate that the list elements are
strings.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tencentcloud_clb_attachment.default[0]: Creating...
tencentcloud_clb_attachment.default[1]: Creating...
tencentcloud_clb_attachment.default[0]: Creation complete after 3s [id=#lbl-g84n3gr2#lb-1sp4y4so]
tencentcloud_clb_attachment.default[1]: Creation complete after 8s [id=#lbl-g84n3gr2#lb-1sp4y4so]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
`

### CODE REVIEW

**Comment by gailwang** [comment]
delete:
`data.tencentcloud_instances.default: Refreshing state...
tencentcloud_clb_attachment.default[0]: Refreshing state... [id=#lbl-g84n3gr2#lb-1sp4y4so]
tencentcloud_clb_attachment.default[1]: Refreshing state... [id=#lbl-g84n3gr2#lb-1sp4y4so]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # tencentcloud_clb_attachment.default[0] will be destroyed
  - resource "tencentcloud_clb_attachment" "default" {
      - clb_id        = "lb-1sp4y4so" -> null
      - id            = "#lbl-g84n3gr2#lb-1sp4y4so" -> null
      - listener_id   = "lbl-g84n3gr2" -> null
      - protocol_type = "TCP" -> null

      - targets {
          - instance_id = "ins-bvpf6j4w" -> null
          - port        = 80 -> null
          - weight      = 10 -> null
        }
      - targets {
          - instance_id = "ins-fa2damgc" -> null
          - port        = 80 -> null
          - weight      = 10 -> null
        }
    }

  # tencentcloud_clb_attachment.default[1] will be destroyed
  - resource "tencentcloud_clb_attachment" "default" {
      - clb_id        = "lb-1sp4y4so" -> null
      - id            = "#lbl-g84n3gr2#lb-1sp4y4so" -> null
      - listener_id   = "lbl-g84n3gr2" -> null
      - protocol_type = "TCP" -> null

      - targets {
          - instance_id = "ins-bvpf6j4w" -> null
          - port        = 80 -> null
          - weight      = 10 -> null
        }
      - targets {
          - instance_id = "ins-fa2damgc" -> null
          - port        = 80 -> null
          - weight      = 10 -> null
        }
    }

Plan: 0 to add, 0 to change, 2 to destroy.


Warning: Quoted type constraints are deprecated

  on variable.tf line 3, in variable "instance_ip":
   3:   type = "list"

Terraform 0.11 and earlier required type constraints to be given in quotes,
but that form is now deprecated and will be removed in a future version of
Terraform. To silence this warning, remove the quotes around "list" and write
list(string) instead to explicitly indicate that the list elements are
strings.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

tencentcloud_clb_attachment.default[0]: Destroying... [id=#lbl-g84n3gr2#lb-1sp4y4so]
tencentcloud_clb_attachment.default[1]: Destroying... [id=#lbl-g84n3gr2#lb-1sp4y4so]
tencentcloud_clb_attachment.default[0]: Destruction complete after 3s
tencentcloud_clb_attachment.default[1]: Destruction complete after 3s
`

**Comment by gailwang** [comment]
tf file:
`data "tencentcloud_instances" "default" {
	vpc_id = "vpc-h70b6b49"
	project_id = 1153961
}

resource "tencentcloud_clb_attachment" default {
  count = length(var.instance_ip)
  clb_id      = "lb-1sp4y4so"
  listener_id = "lbl-g84n3gr2"

  targets {
    instance_id = ["ins-fa2damgc","ins-bvpf6j4w"][count.index]
    port        = 80
    weight      = 10
  }
}
`

File: tencentcloud/resource_tc_clb_attachment.go [hunk 1]
```diff
@@ -233,7 +233,16 @@ func resourceTencentCloudClbServerAttachementRemove(d *schema.ResourceData, meta
 		requestId := ""
 		response, e := meta.(*TencentCloudClient).apiV3Conn.UseClbClient().DeregisterTargets(request)
```
Comment (likexian) [inline tencentcloud/resource_tc_clb_attachment.go:234]: 为毛不用那个函数 DeleteAttachmentById

File: tencentcloud/service_tencentcloud_clb.go [hunk 2]
```diff
@@ -506,6 +506,13 @@ func (me *ClbService) DeleteAttachmentById(ctx context.Context, clbId string, li
 	ratelimit.Check(request.GetAction())
 	response, err := me.client.UseClbClient().DeregisterTargets(request)
 	if err != nil {
+		ee, ok := err.(*sdkErrors.TencentCloudSDKError)
+		if !ok {
+			return errors.WithStack(ee)
+		}
+		if ee.GetCode() == "InvalidParameter" {
+			return nil
+		}
```
Comment (likexian) [inline tencentcloud/service_tencentcloud_clb.go:515]: if ok && ee.GetCode() == "InvalidParameter" {
    return nil
}

**Comment by gailwang** [comment]
caution: this `count`para may cause inconsistant result when updating the attachment resouce, This is an known issue of Terraform.

File: tencentcloud/resource_tc_clb_attachment.go (hunk 1 above)
Comment (gailwang) [reply tencentcloud/resource_tc_clb_attachment.go:234]: 和add一样对称比较好看

**Comment by gailwang** [comment]
make testacc TESTARGS=" -run=TestAccTencentCloudClbServerAttachment"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudClbServerAttachment -timeout 120m
?       github.com/terraform-providers/terraform-provider-tencentcloud  [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/gendoc   0.516s [no tests to run]
=== RUN   TestAccTencentCloudClbServerAttachment_tcp
--- PASS: TestAccTencentCloudClbServerAttachment_tcp (103.91s)
=== RUN   TestAccTencentCloudClbServerAttachment_http
--- PASS: TestAccTencentCloudClbServerAttachment_http (70.47s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     175.316s
?       github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity[no test files]
?       github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper      [no test files]
?       github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit   [no test files]

**Pull request merged**